### ISSUE TITLE
Ability to Group Dashboard States

### ISSUE THREAD

**Issue by alanlgirard** [thingsboard/thingsboard#10545]
**Is your feature request related to a problem? Please describe.**
As a user, when creating a complex dashboard with multiple states, it can be challenging to find the correct state when making changes.  The state window is small and there is no way to group states together for easy navigation or to group states that are related.  

**Describe the solution you'd like**
The states should be able to be assigned to a group just like dashboards, customers, and users can.  The dashboard state selection window would then have the ability to show dashboard states for All or based on the Group.  The search would also search the group field for the state.

![image](https://github.com/thingsboard/thingsboard/assets/4032278/9dbeaaf0-b9bf-4543-ab28-600bbc831d95)
 

**Describe alternatives you've considered**
Alternative solutions currently are to name dashboard states in a defined manner such as adding a group prefix so states will be placed in order by "group" and can also be easily searched by "group"

**Additional context**
Example of the issue can be seen in the new solution templates like the supermarket template that has 44 states and multiple layers to the states with multiple popup states that are associated with a state layer. 
![image](https://github.com/thingsboard/thingsboard/assets/4032278/b182c64d-7064-4c99-9498-e7fef059c26a)

**Comment by rpcai** [comment]
I agree that the Dashboard States feature feels very limited, when developing complex, dynamic solutions. 
I would like to see the "Dashboard states" feature get some attention - including making it easier to navigate between states during development 

A tree-view of dashboards and their contained states would be an effective UI for this feature 

![image](https://github.com/user-attachments/assets/4afc3a18-ba31-4c90-a5dc-925ff4db6fbd)

**Comment by mykhailokornieiev** [comment]
Hi. Such a feature request was previously added to our backlog for developers.
Please follow our [Release Notes](https://thingsboard.io/docs/reference/releases/) to see all new additions and fixes to the platform in subsequent releases.

**Comment by mykhailokornieiev** [comment]
Hi. Unfortunately, but the feature request was rejected, since it goes against the design of Thingsboard.
